### PR TITLE
Deprecate navigation areas

### PR DESCRIPTION
**Pull request by adamziel**
## Description

Related to https://github.com/WordPress/gutenberg/issues/36524. This PR deprecates navigation areas as @noisysocks [suggested](https://github.com/WordPress/gutenberg/pull/36604#issuecomment-974916042).

## Test steps

Confirm that all editors load properly, and the navigation block is being saved and rendered as expected.

### CODE REVIEW

**Review by getdave — APPROVED** [review]
Looks good. This is a good way to soft deprecate the block but also ensure it's not in 5.9.

Could you also raise a follow up Issues to remove the block entirely after a couple of versions of Gutenberg Plugin?

File: packages/block-library/src/navigation-area/edit.js [hunk 1]
```diff
@@ -51,6 +52,10 @@ function NavigationAreaBlock( { attributes, setAttributes } ) {
 		[ navigationAreas ]
 	);
 
+	deprecated( 'wp.blockLibrary.NavigationArea', {
+		since: '5.9',
```
Comment (youknowriad) [inline packages/block-library/src/navigation-area/edit.js:56]: Has this block ever landed on Core as stable? If not, we should use a Gutenberg version here and specify a `plugin: "gutenberg"` here as well

Comment (getdave) [reply packages/block-library/src/navigation-area/edit.js:56]: > Has this block ever landed on Core as stable

Not it's not in a stable release of Core only Gutenberg Plugin. I'll spin up a follow up unless @adamziel gets there first.

Thanks for letting us know.

Comment (getdave) [reply packages/block-library/src/navigation-area/edit.js:56]: @youknowriad Here is the followup https://github.com/WordPress/gutenberg/pull/36777

**Comment by talldan** [comment]
@adamziel @getdave Maybe we should make it so that the block isn't shown in the inserter as well. At the moment the block can still be used with the gutenberg plugin active.

**Comment by getdave** [comment]
Yes I really think we need to hide the block. I guess it's only available for backwards compat so let's hide it.

**Comment by adamziel** [comment]
+1 to that, thank you for bringing it up. Here's a PR https://github.com/WordPress/gutenberg/pull/37026.

**Pull request merged**